pandas/io/tests/test_pickle.py

```diff
@@ -217,6 +217,16 @@ def python_unpickler(path):
                         result = python_unpickler(path)
                         self.compare_element(result, expected, typ)
 
+    def test_compression(self):
+        self.data.to_pickle('data.pkl')
```

Comment:
you will need to separate tests these as you can't guarantee the dependencies for the compression.

But yes the general approach is fine.
also pls use the ``tm.ensure_clean()`` to generate the paths